src/app/campaign.service.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { inject, TestBed } from '@angular/core/testing';
+
+import { Campaign } from './campaign.model';
+import { CampaignService } from './campaign.service';
+import { environment } from '../environments/environment';
+
+describe('CampaignService', () => {
+  beforeEach(() => TestBed.configureTestingModule({
+    imports: [ HttpClientTestingModule ],
+    providers: [ CampaignService ],
+  }));
+
+  it('should retrieve campaign details from API', () => {
+    const service: CampaignService = TestBed.get(CampaignService);
+    const httpMock: HttpTestingController = TestBed.get(HttpTestingController);
+
+    const dummyCampaign: Campaign[] = [{
+      video: [],
```

Comment:
For the service test, I think it would make sense to populate all these empty properties even if the app doesn't show them yet, so we can see that serialisation is working in the service and it's ready to use with those properties.